app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  authorize_resource :topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
+
+  def create
+    if create_post
+      redirect_to create_success_path
+    else
+      redirect_to create_fail_path
+    end
+  end
+
+  def update
+    if @post.update_attributes(posts_params)
+      redirect_to update_success_path
+    else
+      redirect_to update_fail_path
+    end
+  end
+
+  def destroy
+    if @post.destroy
+      redirect_to destroy_success_path
+    else
+      redirect_to destroy_fail_path
+    end
+  end
+
+  # Render a new post in a separate page
+  # This can be used when you do not want to introduce javascript to your implementation
+  def reply
+  end
+
+  protected
+
+  # The default redirecting path of all the actions.
+  # You can rewrite these functions to change specific return paths:
+  #   create_success_path, create_fail_path
+  #   update_success_path, update_fail_path
+  #   destroy_success_path, destroy_fail_path
+  #
+  # @return [String] The return path
+  def default_return_path
+    fail NotImplementedError, 'To be implemented by the concrete topic posts controller.'
+  end
+
+  # Create topic subscriptions for related users
+  #
+  # @return [Bool] True if all subscriptions are created successfully.
+  def auto_subscribe_topic
+    fail NotImplementedError, 'To be implemented by the concrete topic posts controller.'
+  end
+
+  # The topic record that posts belong to.
+  def topic
```

Comment:
abstract. Means if you use some Model A acts_as :topic, then this should be A.topic. I think this easier to understand.